synchronizer/synchronizer.go

```diff
@@ -110,41 +121,26 @@ func (s *ClientSynchronizer) Sync() error {
 					continue
 				}
 			}
-			if waitDuration != s.cfg.SyncInterval.Duration {
-				latestsequencedBatchNumber, err := s.etherMan.GetLatestBatchNumber()
-				if err != nil {
-					log.Warn("error getting latest sequenced batch in the rollup. Error: ", err)
-					continue
-				}
-				// Check latest Synced Batch
-				dbTx, err := s.state.BeginStateTransaction(s.ctx)
-				if err != nil {
-					log.Fatalf("error creating db transaction to get latestSyncedBatch. error: %s", err.Error())
-				}
-				latestSyncedBatch, err := s.state.GetLastBatchNumber(s.ctx, dbTx)
-				errC := dbTx.Commit(s.ctx)
-				if errC != nil {
-					log.Errorf("error committing dbTx, err: %s", errC.Error())
-					rollbackErr := dbTx.Rollback(s.ctx)
-					if rollbackErr != nil {
-						log.Fatalf("error rolling back state. RollbackErr: %s, err: %s",
-							rollbackErr.Error(), errC.Error())
-					}
-					log.Fatalf("error committing dbTx, err: %s", errC.Error())
-				}
+			latestSequencedBatchNumber, err := s.etherMan.GetLatestBatchNumber()
+			if err != nil {
+				log.Warn("error getting latest sequenced batch in the rollup. Error: ", err)
+				continue
+			}
+			latestSyncedBatch, err := s.state.GetLastBatchNumber(s.ctx, nil)
+			if err != nil {
+				log.Warn("error getting latest batch synced. Error: ", err)
+				continue
+			}
+			if latestSyncedBatch >= latestSequencedBatchNumber {
+				log.Info("L1 state fully synchronized")
+				err = s.syncTrustedState(latestSyncedBatch)
 				if err != nil {
-					log.Warn("error getting latest batch synced. Error: ", err)
+					log.Warn("error syncing trusted state. Error: ", err)
 					continue
 				}
-				if latestSyncedBatch == latestsequencedBatchNumber {
-					waitDuration = s.cfg.SyncInterval.Duration
-				}
-				if latestSyncedBatch > latestsequencedBatchNumber {
-					log.Fatal("error: latest Synced BatchNumber is higher than the latest Proposed BatchNumber in the rollup")
-				}
```

Comment:
Why do yu delete this check?? this situation will lead to a wrong synchronizarion and I think we should detected

Reply:
This check is not valid anymore since the trusted stated will be ahead of the latest sequenced batch in the L1.